asmrun/backtrace_prim.c

```diff
@@ -67,6 +67,14 @@ frame_descr * caml_next_frame_descriptor(uintnat * pc, char ** sp)
   }
 }
 
+int caml_alloc_backtrace_buffer(){
+  Assert(caml_backtrace_pos == 0);
```

Comment:
... And no buffer had been allocated before. So you probably need two steps:

1) Capture a first backtrace.
2) Create a new thread, restore the backtrace in this new context.